apps/homescreen/js/app.js

```diff
@@ -435,6 +453,21 @@ const SETTINGS_VERSION = 0;
         handleRoleChange(icon.app, container);
       } else {
         icon.bookmark = appOrBookmark;
+
+        IconsHelper.getIconBlob(icon.bookmark.url, this.iconSize,
```

Comment:
I would rather this was done in gaia-site-icon. Would you ever want to set a bookmark and not have this behaviour? Also, the icon already knows and caches its best size, which would make the above added block unnecessary. Is there a reason this isn't done there?